tap_github/repository_streams.py

```diff
@@ -1916,6 +1920,911 @@ def query(self) -> str:
+                      reactions_count: totalCount
+                      nodes {
+                        reaction_type: content
+                        reacted_at: createdAt
+                        user {
+                          ... on Actor {
+                            login
+                            avatar_url: avatarUrl
+                            html_url: url
+                            type: __typename
+                          }
+                          ... on User {
+                            node_id: id
+                            id: databaseId
+                            site_admin: isSiteAdmin
+                          }
+                        }
+                      }
+                    }
+                  }
+                }
+              }
+            }
+            rateLimit {
+              cost
+            }
+          }
+        """  # noqa: E501
+
+    schema = th.PropertiesList(
+        # Parent keys
+        th.Property("repo", th.StringType),
+        th.Property("org", th.StringType),
+        th.Property("repo_id", th.IntegerType),
+        th.Property("discussion_id", th.IntegerType),
+        th.Property("discussion_number", th.IntegerType),
+        # Discussion Comments keys
+        th.Property("node_id", th.StringType),
+        th.Property("id", th.IntegerType),
+        th.Property("author", user_object),
+        th.Property("author_association", th.StringType),
+        th.Property("body", th.StringType),
+        th.Property("body_html", th.StringType),
+        th.Property("body_text", th.StringType),
+        th.Property("created_at", th.DateTimeType),
+        th.Property("published_at", th.DateTimeType),
+        th.Property("last_edited_at", th.DateTimeType),
+        th.Property("updated_at", th.DateTimeType),
+        th.Property("created_via_email", th.BooleanType),
+        th.Property("deleted_at", th.DateTimeType),
+        th.Property("includes_created_edit", th.BooleanType),
+        th.Property("is_answer", th.BooleanType),
+        th.Property("is_minimized", th.BooleanType),
+        th.Property("minimized_reason", th.StringType),
+        th.Property("upvote_count", th.IntegerType),
+        th.Property("html_url", th.StringType),
+        th.Property("resource_path", th.StringType),
+        th.Property("editor", user_object),
+        th.Property("replies_count", th.IntegerType),
+        th.Property("reactions_count", th.IntegerType),
+        th.Property("reactions", th.ArrayType(reaction_type_object)),
+    ).to_dict()
+
+
+class DiscussionCommentRepliesStream(GitHubGraphqlStream):
+    """
+    Defines stream fetching replies for each discussion comment from each repository.
+
+    Edits made to replies after extraction are not reflected in the stream.
+    Full refresh is required to see the changes.
+
+    Singer SDK does not support smart pagination for GraphQL,
+    so progress for this stream is not resumable if interrupted.
```

Comment:
```suggestion
    Much like the `use_fake_since_parameter` setting for REST streams,
    this stream extracts data in descending order and we exit early when
    we've seen records that go back as far as we need. Since this object
    doesn't offer an orderBy field, we use tail-first pagination to extract
    in descending order. A side effect of this is that within each page the
    records are still ascending order, so some special handling is needed.
    
    Note that this stream is not resumable if interrupted.
```